maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/inbound/messaging/IndexingMessagesStreamListener.java

```diff
@@ -1,107 +1,120 @@
 package bio.overture.maestro.app.infra.adapter.inbound.messaging;
 
 import bio.overture.maestro.domain.api.Indexer;
+import bio.overture.maestro.domain.api.exception.FailureData;
 import bio.overture.maestro.domain.api.message.*;
+import io.vavr.Tuple2;
 import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.jetbrains.annotations.NotNull;
 import org.springframework.cloud.stream.annotation.EnableBinding;
 import org.springframework.cloud.stream.annotation.Input;
 import org.springframework.cloud.stream.annotation.StreamListener;
 import org.springframework.cloud.stream.messaging.Sink;
 import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
+import java.util.function.Function;
+
 @Slf4j
 @EnableBinding(Sink.class)
 public class IndexingMessagesStreamListener {
 
-    private Indexer indexer;
+    private final Indexer indexer;
 
     public IndexingMessagesStreamListener(Indexer indexer) {
```

Comment:
add `@NonNull`

Reply:
you can add it, but for spring it validates beans, by default all beans are required and it can't initialize bean with null

Reply:
Ignore my comment, it's good to add it in case we initialize manually for testing